src/netex_idf/stops.rs

```diff
@@ -128,6 +128,37 @@ fn load_coords(quay: &Element) -> Result<(f64, f64)> {
     Ok((coords[0].parse()?, coords[1].parse()?))
 }
 
+use geo::algorithm::centroid::Centroid;
+use geo_types::MultiPoint;
+
+fn update_stop_area_coords(
+    stop_areas: &mut CollectionWithId<StopArea>,
+    stop_points: &CollectionWithId<StopPoint>,
+) {
+    let mut updated_stop_areas = stop_areas.take();
```

Comment:
I think another possibility would be to `.collect()` the `Idx<StopArea>` in a `Vec` then iterate over this `Vec` and use `.get_mut` on the collection `stop_areas`. This means allocating space for a list of `Idx<StopArea>` but avoids to destroy the original `CollectionWithId`.

If I understand correctly, `.take()` is basically switching some pointers (which is cheap), but then at the end, you have to build again the entire map of the `CollectionWithId` (by calling `CollectionWithId::new()`). The good point though is that the inner `Vec` is never destroyed nor reallocated.

I'm not sure which one is best.

Reply:
it would work but I think with `take` there are less operations allocation
I don't allocation memory with `collect` and I don't iteration on Idxs to find a stop area.